test/integration/InvoiceTest.py

```diff
@@ -77,24 +77,97 @@ def test_invoice(self):
+            {
+                "invoiceId": invoice.id,
+                "amount":{
+                    "value":"11.00",
+                    "currency":"EUR"
+                }
+            },
+            {
+                "invoiceId": invoice2.id,
+                "amount":{
+                    "value":"10.00",
+                    "currency":"EUR"
+                }
+            }
+        ])
+        self.assertEqual(invoice_payment.invoicePayments[0].invoiceId, invoice2.id)
+
+        # Test - Update Invoice Payment
+        update_payment = self.client.invoice_payment.update(
+            {
+                "invoiceId": invoice2.id,
+                "invoiceLineId": invoice2.lines[0]['id'],
+                "paymentId": invoice_payment.paymentId,
+                "amount":{
+                    "value":"21.00",
+                    "currency":"EUR"
+                }
+            }
+        )
+        self.assertTrue(update_payment)
+
+        # Test - Search for Invoice Payments
+        invoice_payments = self.client.invoice_payment.search([invoice_payment.paymentId])
+        self.assertTrue(
+            (invoice_payments[0].invoiceId == invoice_with_lines.id) 
+            or 
+            (invoice_payments[0].invoiceId == invoice2.id))
+
+        # Test - Delete Invoice Payment
+        del_inv_payment = self.client.invoice_payment.delete(invoice_payment.paymentId, [
+            invoice_with_lines.lines[0]['id']
+        ])
+        self.assertTrue(del_inv_payment)
+
+        # Assert that the payment was deleted and can't be found anymore
+        find_payment = self.client.invoice_payment.search([invoice_payment.paymentId])
+        self.assertNotEqual(find_payment[0].invoiceLineId, invoice_with_lines.lines[0]['id'])
+
         # Test - Update Invoice
         ext_id = f"inv-id-{time.time()}"
         invoice = self.client.invoice.update(invoice.id, {
                 "externalId": ext_id
             })
         self.assertEqual(invoice.externalId, ext_id)
 
-        # Assert that the deleted line can't be found
-        self.assertEqual(len(invoice.lines), 1)
+        # Additionally, assert that line item deleted earlier isn't present
+        for line in invoice.lines:
+            self.assertNotEqual(invoice_with_lines.lines[1]['id'], line['id'])
         
         # Test - Get an invoice by id
         fetched_invoice = self.client.invoice.get(invoice.id)
         self.assertEqual(fetched_invoice.externalId, ext_id)
 
         # Test - Delete an Invoice
         response = self.client.invoice.delete([
-            f'{invoice.id}'
+            f'{invoice.id}',
+            f'{invoice2.id}'
         ])
+        
+        # Assert that the deleted invoice is not found anymore
+        try:
+            find_invoice = self.client.invoice.get(invoice.id)
+        except NotFoundException as e:
+            self.assertEqual(e.get_error_array()[0]['code'],'not_found')
```

Comment:
Here, we are trying to fetch the deleted invoice and are catching the exception, then making sure the code is `not_found` - denoting the invoice is actually deleted.